File: 프로그래머스/3/12987. 숫자 게임/숫자 게임.js
function solution(A, B) {
    var answer = 0
    A = A.sort((a, b) => a-b)
    B = B.sort((a,b) => a-b)
    
    let idx = 0
    for(let i=0; i< A.length; i++){
        while(idx < B.length){
            if (A[i] < B[idx]){
                answer += 1
                idx++
                break
            }
            idx++
        }
        
    }
    
    return answer
}

File: 프로그래머스/3/12987. 숫자 게임/숫자 게임.py
import heapq
def solution(A, B):
    heapq.heapify(A)
    heapq.heapify(B)
    answer = 0
    
    while B:
        target = heapq.heappop(A)
        
        while B:
            bPlayer = heapq.heappop(B)
            if (target < bPlayer):
                answer += 1
                break
            
    
    return answer